app/controllers/assessments_controller.rb

```diff
@@ -8,14 +8,28 @@ def create
   end
 
   def show
-    perform_assessment
-  rescue StandardError => e
-    Sentry.capture_exception(e)
-    render json: Decorators::ErrorDecorator.new(e).as_json, status: :unprocessable_entity
+    if assessment_incomplete?
+      render json: Decorators::ErrorDecorator.new(incomplete_message).as_json, status: :unprocessable_entity
+    else
+      begin
+        perform_assessment
+      rescue StandardError => e
+        Sentry.capture_exception(e)
+        render json: Decorators::ErrorDecorator.new(e).as_json, status: :unprocessable_entity
+      end
+    end
```

Comment:
**not a blocker** could move rescue out of if/else statement or even to a rescue_from, but maybe latter is too general?!

```suggestion
    if assessment_incomplete?
      render json: Decorators::ErrorDecorator.new(incomplete_message).as_json, status: :unprocessable_entity
    else
        perform_assessment
     end
   rescue StandardError => e
      Sentry.capture_exception(e)
      render json: Decorators::ErrorDecorator.new(e).as_json, status: :unprocessable_entity
    end
```

```
```